### PR TITLE
chore: refine logging in pkg time

### LINKED ISSUE #2889: Refine logging in `pkg/time`, refactor these functions as methods of a Struct, then move package level variable `log` as a field
> Please leave comments with your idea if you are interested in this issue. And comment `/assign` to operate self-assign before you work on it.

### PR DESCRIPTION
**Pull request by STRRL**
Signed-off-by: STRRL <str_ruiling@outlook.com>

### What problem does this PR solve?

close #2889

### What's changed and how it works?

- add fields logger for `FakeImage`, `SkewClockGetTime`, `SkewGetTimeOfDay`
- update related function signatures

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [x] release-2.1
  - [ ] release-2.0

### Checklist

Tests

- [x] Unit test
- [ ] E2E test
- [ ] No code
- [x] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

**Comment by STRRL** [comment]
/cc 
@Hexilee @Andrewmatilde

**Comment by STRRL** [comment]
PTAL

/cc @Andrewmatilde @iguoyr

**Review by Hexilee — APPROVED** [review]
LGTM

File: pkg/time/asset.go [hunk 1]
```diff
@@ -95,16 +95,22 @@ func LoadFakeImageFromEmbedFs(filename string, symbolName string) (*FakeImage, e
...
 		}
 	}
-	return &fakeImage, nil
+	result := NewFakeImage(
+		symbolName,
+		imageContent,
+		imageOffset,
+		logger,
+	)
+	return result, nil
```
Comment (iguoyr) [inline pkg/time/asset.go:115]:
```suggestion
	return NewFakeImage(
		symbolName,
		imageContent,
		imageOffset,
		logger,
	), nil
```

**Comment by STRRL** [comment]
Updated! PTAL @iguoyr

**Review by iguoyr — APPROVED** [review]
LGTM

**Comment by STRRL** [comment]
> cherry pick to release-2.1 failed

Waiting for #2991

**Pull request merged**